src/test/llm/output_parsers/qwen3coder_output_parser_test.cpp

```diff
@@ -0,0 +1,628 @@
+};
+
+TEST_P(Qwen3CoderOutputParserParametrizedTest, TestJustParserImplWithVariousArgumentTypes) {
+    const std::string& toolName = std::get<0>(GetParam());
+    const std::string& argName = std::get<1>(GetParam());
+    const std::string& paramValue = std::get<2>(GetParam());
+    const std::string& expectedArguments = std::get<3>(GetParam());
+    const std::string input = R"(
+<tool_call>
+<function=)" + toolName +
+                              R"(>
+<parameter=)" + argName + R"(>
+)" + paramValue + R"(
+</parameter>
+</function>
+</tool_call>)";
+    auto content = input;
+    ovms::Qwen3CoderToolParserImpl parser(toolsParametersTypeMap);
+    auto callsOpt = parser.parseChunk(content);
+    ASSERT_TRUE(callsOpt.has_value());
+    ToolCalls_t& calls = callsOpt.value();
+    auto status = parser.removeToolCallsFromContentIfNeeded(content);
+    EXPECT_TRUE(status.ok()) << status.string();
+    EXPECT_EQ(calls.size(), 1) << input;
+    EXPECT_EQ(calls[0].name, toolName);
+    EXPECT_EQ(calls[0].arguments, expectedArguments) << input;
+    EXPECT_EQ(parser.getCurrentState(), ovms::Qwen3CoderToolParserImpl::State::Content) << input;
+    EXPECT_EQ(parser.getLastProcessedPosition(), input.find("</tool_call>") + std::string("</tool_call>").size()) << input;
+    EXPECT_EQ(content, "\n");
+}
+
+const std::vector<std::tuple<std::string, std::string, std::string, std::string>> paramValueAndExpectedArgumentsVec = {
+    {"string_tool", "arg1", "value1", "{\"arg1\": \"value1\"}"},
+    {"int_tool", "arg1", "42", "{\"arg1\": 42}"},
+    {"float_tool", "arg1", "52.32", "{\"arg1\": 52.32}"},
+    {"bool_tool", "arg1", "true", "{\"arg1\": true}"},
+    {"bool_tool", "arg1", "false", "{\"arg1\": false}"},
+    {"bool_tool", "arg1", "True", "{\"arg1\": True}"},
+    {"bool_tool", "arg1", "False", "{\"arg1\": False}"},
+    {"object_tool", "arg1", "{\"a\": 1, \"b\": {\"c\": \"asd\"}}", "{\"arg1\": {\"a\": 1, \"b\": {\"c\": \"asd\"}}}"},
+    {"list_tool", "arg1", "[1, 2, 3]", "{\"arg1\": [1, 2, 3]}"},
+    {"list_tool", "arg1", "[\"a\", \"b\", \"c\"]", "{\"arg1\": [\"a\", \"b\", \"c\"]}"},
+    {"object_tool", "arg1", "[{\"a\": 1}, {\"b\": 2}]", "{\"arg1\": [{\"a\": 1}, {\"b\": 2}]}"}};
+
+INSTANTIATE_TEST_SUITE_P(
+    Qwen3CoderOutputParserParametrizedTestInstance,
+    Qwen3CoderOutputParserParametrizedTest,
+    ::testing::ValuesIn(paramValueAndExpectedArgumentsVec),
+    [](const ::testing::TestParamInfo<Qwen3CoderOutputParserParametrizedTest::ParamType>& info) {
+        std::string name = std::get<0>(info.param) + "_" + std::get<2>(info.param);
+        // Replace non-alphanumeric characters with underscore
+        std::replace_if(name.begin(), name.end(), [](char c) { return !std::isalnum(c); }, '_');
+        // Limit length to 30 characters
+        if (name.length() > 30) {
+            name = name.substr(0, 30);
+        }
```

Comment:
Why is it needed? Not sure if it's safe in case we add more cases that might be similar.

Reply:
It's just to avoid very long names in case of JSON objects. Gtest wouldn't allow test with the same name to be generated anyway.